src/proof/proof-service.ts

```diff
@@ -756,24 +780,28 @@ export class ProofService implements IProofService {
     }
 
     let operator = 0;
-    const values: bigint[] = new Array<bigint>(64).fill(BigInt(0));
     const [key, value] = fieldReqEntries[0];
     if (!QueryOperators[key]) {
       throw new Error(`operator is not supported by lib`);
     }
     operator = QueryOperators[key];
+
+    query.operator = operator;
+
+    return { query, fieldName, rawValue: value };
+  }
+
+  async transformQueryValueToBigInts(value: unknown, ldType: string): Promise<bigint[]> {
+    const values: bigint[] = new Array<bigint>(64).fill(BigInt(0));
+
     if (Array.isArray(value)) {
       for (let index = 0; index < value.length; index++) {
-        values[index] = BigInt(value[index]);
+        values[index] = await Merklizer.hashValue(ldType, value);
```

Comment:
@vmidyllic shouldn't this convert `value[index]`?

Reply:
thank you. Fixed in 1.0.0-beta.15